crates/nu-command/src/platform/input/input_.rs

```diff
@@ -75,139 +80,97 @@ impl Command for Input {
+                                {
+                                    if k.modifiers == KeyModifiers::CONTROL && c == 'c' {
+                                        crossterm::terminal::disable_raw_mode()?;
+                                        return Err(ShellError::IOError("SIGINT".to_string()));
                                     }
-                                    KeyCode::Enter => {
+                                    continue;
+                                }
+
+                                if let Some(byte_until) = byte_until {
+                                    if c == byte_until as char {
                                         break;
                                     }
-                                    _ => continue,
                                 }
+                                buf.push(c);
+                            }
+                            KeyCode::Backspace => {
+                                let _ = buf.pop();
+                            }
+                            KeyCode::Enter => {
+                                break;
                             }
                             _ => continue,
-                        },
-                        Ok(_) => continue,
-                        Err(event_error) => {
-                            crossterm::terminal::disable_raw_mode()?;
-                            return Err(event_error.into());
                         }
                     }
+                    _ => continue,
+                },
+                Ok(_) => continue,
+                Err(event_error) => {
+                    crossterm::terminal::disable_raw_mode()?;
+                    return Err(event_error.into());
                 }
-                crossterm::terminal::disable_raw_mode()?;
-                return Ok(Value::String {
-                    val: buf,
-                    span: call.head,
-                }
-                .into_pipeline_data());
             }
-
-            // Just read a normal line of text, and trim the newline at the end
-            let input = std::io::stdin().read_line(&mut buf);
-            if buf.ends_with('\n') {
-                buf.pop();
-                if buf.ends_with('\r') {
-                    buf.pop();
-                }
-            }
-
-            match input {
-                Ok(_) => Ok(Value::String {
-                    val: buf,
-                    span: call.head,
-                }
-                .into_pipeline_data()),
-                Err(err) => Err(ShellError::IOError(err.to_string())),
+            if !suppress_output {
+                // clear the current line and print the current buffer
+                execute!(
+                    std::io::stdout(),
+                    terminal::Clear(ClearType::CurrentLine),
+                    cursor::MoveToColumn(0),
+                    Print(buf.to_string()),
+                )?;
```

Comment:
This nukes the prompt if there is one.

Reply:
https://github.com/nushell/nushell/pull/10150/commits/ea6f48ce442f787b83586b746ae25f303ad84d6e